File: model-client/src/main/java/org/modelix/model/api/PNodeResolveContext.java
package org.modelix.model.api;

public class PNodeResolveContext implements INodeResolveContext {
  private IBranch branch;

  public PNodeResolveContext(IBranch branch) {
    this.branch = branch;
  }

  public IBranch getBranch() {
    return branch;
  }

  @Override
  public INode resolve(INodeReference ref) {
    if (ref instanceof PNodeReference) {
      return new PNodeAdapter(((PNodeReference) ref).getId(), branch);
    } else {
      return null;
    }
  }
}

File: model-client/src/main/java/org/modelix/model/api/PNodeResolveContext.kt
package org.modelix.model.api

class PNodeResolveContext(val branch: IBranch) : INodeResolveContext {

    override fun resolve(ref: INodeReference?): INode? {
        return if (ref is PNodeReference) {
            PNodeAdapter(ref.id, branch)
        } else {
            null
        }
    }
}
